pwiz_tools/Skyline/FileUI/PeptideSearch/MatchModificationsControl.cs

```diff
@@ -116,38 +130,73 @@ public bool Initialize(SrmDocument document)
 
         public SrmSettings AddCheckedModifications(SrmDocument document)
         {
-            if (modificationsListBox.CheckedItems.Count == 0)
-                return document.Settings;
+            if (modificationsListBox.CheckedItems.Count == 0 && !ImportPeptideSearch.IsDDASearch)
+                {
+                if (!ImportPeptideSearch.IsDDASearch)
+                    return document.Settings;
+                else
+                {
+                    document.Settings.PeptideSettings.Modifications.StaticModifications.Clear();
+                    return document.Settings;
+                }
+            }
 
             // Find checked static mods
-            var newStructuralMods = (from mod in ImportPeptideSearch.MatcherPepMods.StaticModifications
+            List<StaticMod> structuralMods;
+            List<StaticMod> newHeavyMods; 
+      if (ImportPeptideSearch.IsDDASearch)
+            {
+                structuralMods = (from mod in Settings.Default.StaticModList
+                    from ListBoxModification checkedMod in modificationsListBox.CheckedItems
+                    where mod.Equivalent(checkedMod.Mod)
+                    select mod).ToList();
+        //TODO Check for amanda
+        
+                newHeavyMods = new List<StaticMod>();
+
+            }
+            else
+            {
+            
+            structuralMods = (from mod in ImportPeptideSearch.MatcherPepMods.StaticModifications
                 from ListBoxModification checkedMod in modificationsListBox.CheckedItems
                 where mod.Equivalent(checkedMod.Mod)
                 select mod).ToList();
 
             // Find checked heavy mods
-            var newHeavyMods = (from mod in ImportPeptideSearch.MatcherHeavyMods
+            newHeavyMods = (from mod in ImportPeptideSearch.MatcherHeavyMods
                 from ListBoxModification checkedMod in modificationsListBox.CheckedItems
                 where mod.Equivalent(checkedMod.Mod)
                 select mod).ToList();
+          }
 
             // Update document modifications
             return ImportPeptideSearch.AddModifications(document,
-                new PeptideModifications(newStructuralMods, new[] {new TypedModifications(IsotopeLabelType.heavy, newHeavyMods)}));
+                new PeptideModifications(structuralMods, new[] {new TypedModifications(IsotopeLabelType.heavy, newHeavyMods)}));

```

Comment:
The way mods are presented in the wizard seems backward to me for DDA. I expect to first be asked about what mods to include in the search, and later asked whether to import them into the document from the library. At least the caption needs to change because it's talking about search results when the search hasn't been conducted yet. :)
![image](https://user-images.githubusercontent.com/5011937/82486968-ab2a9080-9aab-11ea-8343-d487a9e8cbe5.png)

Reply:
true! changed the caption!
asking the user which modifications shall be imported is still missing (added a todo). do you think we should show again the MatchModificationsControl?